### PR TITLE
Rename autopara env vars and get rid of "tasks" at expyre level

### LINKED ISSUE #67: rename `chunksize` and `job_chunksize`
Some ideas: 

`job_chunksize` -> `items_per_job`

`chunksize` ->  
* serial_chunk_size 
* serial_chunksize
* num_serial
* num_serial_items
* num_serial_ops
* num_operations_in_serial
* items_in_pool

### PR DESCRIPTION
**Pull request by bernstei**
Rename internal and env variables as per pre-docathon discussion.  

No more "_AUTOPARA_".  

npool -> num_python_subprocesses, chunksize -> num_inputs_per_python_subprocess, and job_chunksize -> num_inputs_per_queued_job.  

probably more.

Needs to use https://github.com/libAtoms/ExPyRe/pull/14

closes #67

### CODE REVIEW

File: tests/test_remote_run.py [hunk 1]
```diff
@@ -138,14 +138,14 @@ def do_generic_calc(tmp_path, sys_name, monkeypatch):
...
-    monkeypatch.setenv('WFL_AUTOPARA_REMOTE_NO_MARK_PROCESSED', '1')
+    monkeypatch.setenv('WFL_EXPYRE_NUM_O_MARK_PROCESSED', '1')
 
     t0 = time.time()
     results = generic.run(inputs=ci, outputs=co, calculator=calc)
     dt = time.time() - t0
     print('remote parallel calc_time', dt)
 
-    monkeypatch.delenv('WFL_AUTOPARA_REMOTE_NO_MARK_PROCESSED')
+    monkeypatch.delenv('WFL_EXPYRE_NUM_O_MARK_PROCESSED')
```
Comment (gabor1) [inline tests/test_remote_run.py:140]: what's NUM_O ?

**Comment by gelzinyte** [comment]
we talked about making reports like this
```
Running <function run_op at 0x2b8afaee0f70> with num_python_subprocesses=16, num_inputs_per_python_subprocess=1
```
clearer, is this a good time & PR?

Would it be a matter of renaming functions, like `run_op` here? Or making the function that does the printing find the name by some other means?

File: tests/test_cli_rss.py [hunk 2]
```diff
@@ -113,7 +113,7 @@ def do_full_test(runner, assets_dir, monkeypatch):
 
     # make sure nothing is parallel so things are as deterministic as possible
     monkeypatch.setenv("WFL_DETERMINISTIC_HACK", "1")
-    # monkeypatch.setenv("WFL_AUTOPARA_NPOOL", "0")
+    # monkeypatch.setenv("WFL_NUM_PYTHON_SUBPROCESSES", "0")
     monkeypatch.setenv("OMP_NUM_THREADS", "1")
     monkeypatch.setenv("GAP_FIT_OMP_NUM_THREADS", "1")
```
Comment (gelzinyte) [inline tests/test_cli_rss.py:118]: how about pre-pending `GAP_FIT_OMP_NUM_THREADS` and similar (also for ACE_FIT) with `WFL`?

File: wfl/autoparallelize/remote.py [hunk 3]
```diff
@@ -122,7 +122,7 @@ def do_remotely(remote_info, hash_ignore=[], chunksize=1, iterable=None, outputs
 
     outputspec.end_write()
 
-    if 'WFL_AUTOPARA_REMOTE_NO_MARK_PROCESSED' not in os.environ:
+    if 'WFL_EXPYRE_NUM_O_MARK_PROCESSED' not in os.environ:
```
Comment (gelzinyte) [inline wfl/autoparallelize/remote.py:125]: typo?

File: tests/test_remote_run.py (hunk 1 above)
Comment (bernstei) [reply tests/test_remote_run.py]: typo.  fixed now.

File: README.md [hunk 4]
```diff
@@ -189,18 +189,23 @@ will run on every system defined in `$HOME/.expyre/config.json`.  The
...
-> env EXPYRE_PYTEST_SYSTEMS='(local_sys|remote_sys)' pytest -s --basetemp $HOME/pytest \
-      --runremote --runslow -rxXs
+> env EXPYRE_PYTEST_SYSTEMS='(local_sys|remote_sys)' \
+      WFL_PYTEST_BUILDCELL=path_to_buildcell  \
+      VASP_COMMAND=path_to_vasp VASP_COMMAND_GAMMA=path_to_vasp_gamma PYTEST_VASP_POTCAR_DIR=path_to_potcars \
+      ASE_ORCA_COMMAND=path_to_orca \
+      pytest -s --basetemp $HOME/pytest --runremote --runslow -rxXs
 ```
 
 ## TODO
```
[the remark this answers is not in the document]
Comment (bernstei) [reply README.md:207]: I don't think I follow. If you mean an issue to make the automated tests complete, we may never have a complete test, given that people are quite likely to not have access to some DFT code.

File: tests/test_cli_rss.py (hunk 2 above)
Comment (bernstei) [reply tests/test_cli_rss.py:118]: Good idea.  I'll do that.

**Comment by bernstei** [comment]
> we talked about making reports like this
> 
> ```
> Running <function run_op at 0x2b8afaee0f70> with num_python_subprocesses=16, num_inputs_per_python_subprocess=1
> ```
> 
> clearer, is this a good time & PR?
> 
> Would it be a matter of renaming functions, like `run_op` here? Or making the function that does the printing find the name by some other means?

I'll try to do it via the routine that calls the print statement, so we don't have to rename tons of things.

**Comment by bernstei** [comment]
> I mainly see renames - were there any consequences in workflow of removing "tasks" from Expyre? Or is that just that examples will need to be updated?

Just examples/docs.

> Also documentation that has been merged into main since this PR needs updating - mind if I merge `main` into here and clean that up?

Go ahead and merge docs into main, then I'll merge from main into this branch

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (gelzinyte) [reply README.md:207]: Just thought that some of these TODOs are outdated and those that aren't might be better placed in issues?

File: wfl/autoparallelize/remote.py (hunk 3 above)
Comment (bernstei) [reply wfl/autoparallelize/remote.py]: yes, fixed now I think.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (bernstei) [reply README.md:207]: Aah, good point.  Yes. Do you know enough to open the needed issues, or is it unclear?

File: wfl/autoparallelize/remoteinfo.py [hunk 5]
```diff
@@ -37,19 +37,19 @@ class RemoteInfo:
...
         self.job_name = job_name
         self.resources = copy.deepcopy(resources)
-        self.job_chunksize = job_chunksize
+        self.num_inputs_per_queued_job = num_inputs_per_queued_job
         self.pre_cmds = pre_cmds.copy()
         self.post_cmds = post_cmds.copy()
-        self.env_vars = ["WFL_AUTOPARA_NPOOL=${EXPYRE_NTASKS_PER_NODE}",
-                         "OMP_NUM_THREADS=${EXPYRE_NCORES_PER_TASK}"] + env_vars
+        if all(not var.startswith("WFL_NUM_PYTHON_SUBPROCESSES=") for var in env_vars):
+            self.env_vars = ["WFL_NUM_PYTHON_SUBPROCESSES=${EXPYRE_NUM_CORES_PER_NODE}"] + env_vars
```
[the remark this answers is not in the document]
Comment (gelzinyte) [reply wfl/autoparallelize/remoteinfo.py:52]: wouldn't `self.env_vars` be left undefined? So whatever is expecting some would crash? 

I remember people complaining about `OMP_NUM_THREADS`, but not `WFL_NUM_PYTHON_SUBPROCESSES`? I would say a much more intuitive default would be to parallelise with one core per python subprocess?

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (gelzinyte) [reply README.md:207]: Opened the first one, the third is outdated and I am not entirely sure what the second and fourth mean?

**Comment by gelzinyte** [comment]
> Go ahead and merge docs into main, then I'll merge from main into this branch

docs are in main - will you rename things or should I?

File: tests/test_cli_rss.py (hunk 2 above)
Comment (bernstei) [reply tests/test_cli_rss.py]: done now, I believe.

File: wfl/autoparallelize/remoteinfo.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (bernstei) [reply wfl/autoparallelize/remoteinfo.py]: Can you take a look at the latest, and confirm that you agree that it's fixed now?  If so, just resolve this conversation please.

**Comment by bernstei** [comment]
> I'll try to do it via the routine that calls the print statement, so we don't have to rename tons of things.

Done using `inspect` now.

**Comment by bernstei** [comment]
> > docs are in main - will you rename things or should I?
> 
> I can do it, I guess, since it's so similar to what I already did for the code itself.

Done now.

**Comment by bernstei** [comment]
I think everything we discussed is done.  If someone else can review all the remaining comments and confirm that I didn't miss anything, that'd be great.  

The one thing I still have a tiny worry about is that we replaced `chunksize` with `num_inputs_per_python_subprocess`, but that's not quite right.  It's not the number for each subprocess, but rather the number that's sent _at one time_ to each subprocess.  With its current name, people might assume that it's required that `n_inputs = num_python_subprocesses * num_inputs_per_python_subprocess`, which is not correct.  Comments?

**Comment by bernstei** [comment]
The even more verbose but more precise version would be `num_inputs_per_python_subprocess_call`, or something like that.

**Comment by gelzinyte** [comment]
I think both `num_inputs_per_python_subprocess` and `num_inputs_per_python_subprocess_call` are fine, I'd probably stick with the first one. One, I think the name is detailed enough for people to understand what it does roughly just from the name and for the detail one would go to docstrings anyway. 

Everything else looks great for merging!

**Pull request merged**